doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+; field                               = value
+mac_addr                              = mac_address                   ; global anycast mac MAC address
+
+key                                   = INTF_TABLE:sag_name:ip_prefix ; SAG IP address
+; field                               = value
+scope                                 = global/local/host             ; IP address scope
+family                                = IPv4/IPv6                     ; IP address family
+```
+
+## Orchestration Agent
+Following orchagents shall be modified. Flow diagrams are captured in a later section. 
+
+![SAG arch](./images/sag-db-arch.png)
+
+### SagMgrD
+SagMgrD creates the macvlan kernel interfaces on top of the vlan interface.  
+SagMgrD waits for macvlan interface creation update in STATE_DB and updates the APP_DB INTF_TABLE with the macvlan interface name.
+SagMgrD creates IP address on the SAG interface. 
+SagMgrD gets the INTERFACE Table config to create route entry in SAI from APP_DB updates
+
+### IntfsOrch
+Add SagOrch as a member of IntfsOrch. IntfsOrch creates Router Interfaces based on interface table (INTF_TABLE). For SAG usecase, IntfOrch calls SagOrch API to handle router interface creation.
+
+### PortOrch
+Add SAG related port interface and corresponding attributes.
+Also handle the port referenece count for the SAG port interface.
+
+# SAI
+There are no changes to SAI headers/implementation to support this feature. Existing SAI router interface APIs are used to handle creating SAG interface. SAG needs to configure SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL to true when creating router interface.
+
+# Flows
+The overall data flow diagram is captured below for all TABLE updates. 
+ 
+ ![SAG flows](./images/sag-flows.png)
+
+# CLI
+Commands summary
+
+  - config sag mac_address add <mac_address>
+      - Add global mac address for sag
+  - config sag mac_address del <mac_address>
+      - Remove global mac address for sag
+  - config sag ipv4 enable
+      - Enable the ipv4 knob for sag
+  - config sag ipv4 disable
+      - Disable the ipv4 knob for sag
+  - config sag ipv6 enable
+      - Enable the ipv6 knob for sag
+  - config sag ipv6 disable
+      - Disable the ipv6 knob for sag
+  - config interface sag ip add <vlan_name> <ip_addr>
+      - Create an sag interface and set ip on vlan interface
+  - config interface sag ip del <vlan_name> <ip_addr>
+      - Delete the ip on sag interface
+  - show sag
+      - Show global mac for sag
+```
+Static Anycast Gateway Information
+
+MacAddress         IPv4    IPv6
+-----------------  ------  ------
+00:11:22:33:44:55  enable  enable
+```
+  - show sag ip
+    - Show ipv4 on sag interface
+
+```
+Vlan Interface Name    IPv4 address/mask
+---------------------  -------------------
+Vlan100                1.1.1.254/24
+```
+  - show sag ipv6
+    - Show ipv6 on sag interface
+
+```
+Vlan Interface Name    IPv6 address/mask
```

Comment:
Can we add SAI IP as a filed in 'show vlan brief' ?

Reply:
In the updated HLD, we use the original "show ip interfaces/show ipv6 interfaces" to display the IP address configured on the VLAN interface, no new command is added. 
For SAG enable/disable status on the VLAN interface, it adds a extra column in "show vlan brief", please refer to the updated HLD.